[com_github_nelhage_rules_boost] Disable iostreams deps

OpenUSD uses :multi_index which depends on :serialization which
depends on :iostreams which depends on some compression libraries.

In Drake, we don't expect to use any of those compression features,
so we'll cut away the extra dependencies for simplicity.

Since this is a Drake-specific customization, we don't plan on
upstreaming this patch.


--- boost.BUILD
+++ boost.BUILD
@@ -1131,10 +1131,13 @@
         ":type",
         ":type_traits",
         ":utility",
-        "@com_github_facebook_zstd//:zstd",
-        "@org_bzip_bzip2//:bz2lib",
-        "@org_lzma_lzma//:lzma",
-        "@zlib",
+    ],
+    exclude_src = [
+        "libs/iostreams/src/bzip2.cpp",
+        "libs/iostreams/src/gzip.cpp",
+        "libs/iostreams/src/lzma.cpp",
+        "libs/iostreams/src/zlib.cpp",
+        "libs/iostreams/src/zstd.cpp",
     ],
 )
 
